Add panel component helpers using a lib and railtie

naseberry opened the pull request:
#### What
Add the Govuk Panel component and replace existing markup

#### Ticket
[Add component - Panel](https://dsdmoj.atlassian.net/browse/CBO-1322)

#### Why
Migrating to GDS design system to enable a consistent user experience

#### How
This pattern should make moving to a rails extension
gem easier, and make it easy to add components
in a modular fashion.

The GovukComponent::Railtie class mixes in the
helpers at boot time. see [this good blog on the subject](https://blog.capsens.eu/rails-6-boot-sequence-d289b44d2e94)

To get around the fact that railties need to be
intialized early in the boot sequence the railtie
is `require`'d in the `config/application.rb` (just after
normal Gem `require`ing). Once moved to a gem this can
be removed and would simply work as part of the normal
`Bundler.require(*Rails.groups)` call in that file.

jsugarman commented on config/application.rb:
```diff
@@ -2,11 +2,16 @@
 
 require 'rails/all'
 
+# TODO: test if this is (still) required
```
@naseberry unrelated but noticed this and not sure if we are using [susy]( https://www.oddbird.net/susy/)

naseberry replied:
@jsugarman  Susy is being used by our style sheet, however it is now loaded via npm.

I need to create a task to test and if required remove redundant rails dependencies and configuration related to asset management.

jsugarman replied:
cool, can we remove the comment then

naseberry replied:
@jsugarman yes please

jsugarman commented:
@naseberry are you wanting to get this PR in?

naseberry commented:
> @naseberry are you wanting to get this PR in?
@jsugarman Yes, I will deploy it to an env to be tested

naseberry commented:
@jsugarman would you care to review again. I have made the following changes:

- Updated helper
- Added rspec-html-matchers gem
- Update panel spec
